torchtune/modules/peft/lora.py

```diff
@@ -142,3 +298,29 @@ def _lora_b_init_params(x: nn.Linear) -> None:
     Initialize LoRA B weight to zeros.
     """
     nn.init.zeros_(x.weight)
+
+
+def _swap_lora_linear_with_qat(
+    module: nn.Module,
+    # TODO: make the types Optional[FakeQuantizeConfig] once we
+    # support torchao 0.7+ by default
+    activation_qat_config: Optional["FakeQuantizeConfig"] = None,
+    weight_qat_config: Optional["FakeQuantizeConfig"] = None,
+) -> None:
+    """
+    Swap all `LoRALinear`s in the model with `QATLoRALinear`.
+    """
+    for name, child in module.named_children():
```

Comment:
Kinda a similar comment here: why do we do this recursively instead of just bottoms-up iteration over named_modules? I guess this way is more robust?

Reply:
this one's easier with recursion because we can `setattr` directly on the submodule